src/erd-script.sql

```diff
@@ -0,0 +1,255 @@
+CREATE SCHEMA IF NOT EXISTS pet_haven_schema;
+
+SET
+  SEARCH_PATH TO pet_haven_schema;
+
+-- Role Table
+CREATE TABLE roles (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(50) UNIQUE NOT NULL
+);
+
+-- Insert default roles
+INSERT INTO
+  roles (name)
+VALUES
+  ('Admin'),
+  ('Manager'),
+  ('Client');
+
+-- Users Table
+CREATE TABLE users (
+  id BIGINT PRIMARY KEY,
+  first_name VARCHAR(255) NOT NULL,
+  last_name VARCHAR(255) NOT NULL,
+  email VARCHAR(255) UNIQUE NOT NULL,
+  password_hash TEXT NOT NULL,
+  role_id INT NOT NULL,
+  avatar_url VARCHAR(255),
+  is_enabled BOOLEAN NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  FOREIGN KEY (role_id) REFERENCES roles (id)
+);
+
+-- Auth Tokens Table
+CREATE TABLE auth_tokens (
+  id SERIAL PRIMARY KEY,
+  user_id INT NOT NULL,
+  token TEXT NOT NULL,
+  expires_at TIMESTAMP NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  FOREIGN KEY (user_id) REFERENCES users (id)
+);
+
+-- Activation Code Table
+CREATE TABLE activation_code (
+  id SERIAL PRIMARY KEY,
+  user_id INT NOT NULL,
+  key VARCHAR(255),
+  expiration_date TIMESTAMP,
+  FOREIGN KEY (user_id) REFERENCES users (id)
+);
+
+-- Categories Table
+CREATE TABLE categories (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  created_by VARCHAR(255),
+  last_modified_by VARCHAR(255)
```

Comment:
since these are audit columns, it should be good to also make them reference the users table (imagine having changes by an unknown user!)